backbone.js

```diff
@@ -1621,11 +1620,12 @@
       // Normalize the fragment.
       fragment = this.getFragment(fragment || '');
 
-      // Don't include a trailing slash on the root.
+      // Append a trailing slash if needed.
       var root = this.root;
-      if (fragment === '' || fragment.charAt(0) === '?') {
-        root = root.slice(0, -1) || '/';
+      if (fragment && this._usePushState) {
+        if (root.charAt(root.length - 1) !== '/' && fragment.charAt(0) !== '?') root += '/';
       }
```

Comment:
I think I'd flip this on it's head and prepend the slash to `fragment` when needed.

```js
if (this._usePushState && fragment.charAt(0) !== '?') fragment = '/' + fragment;
```